core/src/main/resources/org/apache/spark/ui/static/historypage-template.html

```diff
@@ -77,7 +77,12 @@
       <td><span title="{{duration}}" class="durationClass">{{duration}}</span></td>
       <td>{{sparkUser}}</td>
       <td>{{lastUpdated}}</td>
-      <td><a href="{{uiroot}}/api/v1/applications/{{id}}/{{num}}/logs" class="btn btn-info btn-mini">Download</a></td>
+      <td>
+        <a href="{{uiroot}}/api/v1/applications/{{id}}/{{num}}/logs" class="btn btn-info btn-mini">Download</a>
+        <a href="{{uiroot}}/logs/delete/?appId={{id}}&attemptId={{attemptId}}"
+           onclick="if (window.confirm('Are you sure you want to delete the Event Log for {{id}} from the server?\n\nThis can not be undone!')) { this.parentNode.submit(); return true; } else { return false; }"
```

Comment:
I had trouble moving this js code into a separate file and still have it function as expected. If leaving it here is a problem I can try replicate its functionality by other methods. This was based on the kill job/stages code.